samza-core/src/main/java/org/apache/samza/table/caching/CachingTable.java

```diff
@@ -105,96 +100,208 @@ public void init(SamzaContainerContext containerContext, TaskContext taskContext
     tableMetricsUtil.newGauge("req-count", () -> requestCount());
   }
 
+  /**
+   * Lookup the cache and return the keys that are missed in cache
+   * @param keys keys to be looked up
+   * @param records result map
+   * @return list of keys missed in the cache
+   */
+  private List<K> lookupCache(List<K> keys, Map<K, V> records) {
+    List<K> missKeys = new ArrayList<>();
+    records.putAll(cache.getAll(keys));
+    keys.forEach(k -> {
+        if (!records.containsKey(k)) {
+          missKeys.add(k);
+        }
+      });
+    return missKeys;
+  }
+
   @Override
   public V get(K key) {
+    try {
+      return getAsync(key).get();
+    } catch (InterruptedException e) {
+      throw new SamzaException(e);
+    } catch (Exception e) {
+      throw (SamzaException) e.getCause();
```

Comment:
Maybe do a pass and change the exception handling logic to not inspect "e.getCause()".